### DISCUSSION TITLE
AutoComplete - suggestions exist, but not displaying the overlay

### DISCUSSION THREAD

**Discussion by amillion3** [primefaces/community#533] [PrimeVue]
I have ran into this on 2 totally different codebases now. Unsure if this a bug or (likely) just me missing a crucial step. 

On the latest version of PrimeVue (3.41.0) using Vue 3 (composition API). Here is my PrimeVue AutoComplete component named: `TbsAutoComplete`

```
<template>
  <AutoComplete
    v-model="searchText"
    :suggestions="items"
    option-label="name"
    data-key="code"
    @complete="search"
  />
</template>

<script setup>
import {ref} from "vue"
import AutoComplete from "primevue/autocomplete"
import SearchService from "@/services/searchService.js"

const searchText = ref("")
const items = ref([])

const search = async (event) => {
  await SearchService.promisePerformSearch(searchText.value)
    .then(response => {
      response.forEach(res => {
        items.value.push({name: res.ResultsDescription, code: res.ProjectId})
      })
      return true
    })
    .catch(error => { ... })
}
</script>
```

The API call is made and the data is returned as the `response` array. This is not the issue.

The `items` array gets populated with the correct values:
![image](https://github.com/primefaces/community/assets/20422019/3f811b68-dbf2-4144-8d07-4ce232edb6be)

The actual `AutoComplete` component gets in the list of suggestions as well:
![image](https://github.com/primefaces/community/assets/20422019/b255b9bf-24d1-4933-8ad1-ced0a92dd265)

I think what's happening is that the `@complete` handler is never resolving or finishing, or triggering something. If I examine the `AutoComplete` component, I see a couple issues:
![image](https://github.com/primefaces/community/assets/20422019/843358ba-acdf-40b6-8584-d032d1d6b181)

If I manually toggle `overlayVisible` to true and `searching` to false, I see the AutoComplete options appear!
![image](https://github.com/primefaces/community/assets/20422019/880d79a8-81c2-474e-befe-ed135c1ef004)

Any thoughts? I would appreciate any insights or workarounds. Even if I can manually trigger the `overlayVisible` and `searching` values, I can live with that.

**Comment by Rakasch** [comment]
Can you replicate it in stackblitz? Seems to work fine for me.
https://stackblitz.com/edit/q8ih4f?file=src%2FApp.vue

In worst case, you can put a `ref` on the autocomplete, access the components internal data and change `overlayVisible`, `searching` as you like.
This is obviously not the preferred solution, as you create a tight coupling and are more likely to have problems with updating if the autocomplete component changes internally.

As an aside:
In the primevue examples on lazy loading stuff,  they always collect everything in a tmp array and then update the items array only once.
That's why I came to the assumption, that maybe it's not a good idea to push each item individually.
I'm seriously not that experianced with vuejs, maybe you would trigger the reactive updates on each push to items?
Or does vuejs only update once each tick/cycle?
